slides/src/content/template-forms.md

```diff
@@ -0,0 +1,266 @@
+<!-- .slide: data-background="../content/images/title-slide.jpg" -->
+
+## Building Applications with Angular
+
+# Template-Driven Forms
+
+---
+
+## Roadmap
+
+1. How does Angular support forms?
+1. How do I create a template driven form?
+1. How do I handle form data?
+1. How do I validate form data?
+1. How do I display error messages?
+
+---
+
+## Overview
+
+- Template Driven Forms: controls and validation rules defined in the template with directives
+- Reactive Forms: controls and validation rules defined in the component class or service
+  - Explore these in the next section
+
+---
+
+## Creating Template Driven Forms
+
+- Import `FormsModule` from `@angular/forms` into `app.module.ts`
+- Allow us to use all the features of template-driven forms
+
+_src/app/app.module.ts_
+```ts
+…existing imports…
+import { FormsModule } from '@angular/forms';
+
+@NgModule({
+  …as before…
+  imports: [
+    BrowserModule,
+    HttpModule,
+    FormsModule        // added
+  ],
+  …as before…
+})
+export class AppModule { }
+```
+
+---
+
+## Handling Form Data
+
+- Use `ngModel` and `ngForm` directives to handle form data
+- `ngForm` provides properties to get information from the form like `value` and `valid`
+- `ngModel` provides the same property for individual fields
+
+_src/app/generic-input/generic-input.component.html_
+```html
+<form #newItemForm="ngForm">
+  <p>New Item: <input name="newItem" ngModel></p>
+</form>
+```
+
+<!-- example: https://plnkr.co/edit/sdVst5yprdQUgQctQz9p?p=preview -->
+
+---
+
+## Handling Form Data
+
+- Every `<form>` element is automatically enhanced by the `NgForm` directive
+- `newItemForm` is a template variable that holds a reference to the `NgForm` directive instance
+- `ngModel` turns a form field into a `FormControl` using the element `name` property…
+- …and registers it with the form
+- Every control that registers with the `ngForm` automatically appears in the `form.value` property
+
+---
+
+## NgForm
+
+`NgForm` instances have several useful properties:
+
+- `newItemForm.value`: all form values
+- `newItemForm.valid`: whether the form is valid
+- `newItemForm.value.newItem`: the value of the `newItem` field
+
+---
+
+## NgModel
```

Comment:
Since there are two sections for NgModel, should this be 1/2 and the following one 2/2?